File: src/main/java/info/devels/jtte/blocks/BlockBeacon.java
package info.devels.jtte.blocks;

import cofh.api.block.IBlockInfo;
import cofh.api.core.IInitializer;
import cofh.api.tileentity.ITileInfo;
import cofh.core.block.BlockCoFHBase;
import cofh.lib.util.helpers.ItemHelper;
import cpw.mods.fml.common.registry.GameRegistry;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import info.devels.jtte.JTTE;
import info.devels.jtte.entities.TileEntityBeacon;
import net.minecraft.block.Block;
import net.minecraft.block.ITileEntityProvider;
import net.minecraft.block.material.Material;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.IChatComponent;
import net.minecraft.world.IBlockAccess;
import net.minecraft.world.World;
import net.minecraftforge.common.util.ForgeDirection;

import java.util.ArrayList;
import java.util.List;


public class BlockBeacon extends Block implements ITileEntityProvider, IBlockInfo, IInitializer {
    public static ItemStack blockBeacon;

    public BlockBeacon() {
        super(Material.rock);
        setHardness(5.0F);
        setResistance(100F);
        setStepSound(soundTypeStone);
        setCreativeTab(JTTE.tab);
        setBlockTextureName("obsidian");
        setBlockName("jtte.beacon");

        setBlockBounds(0F, 0F, 0F, 1F, 1.8F, 1F);
    }

    @Override
    public int getLightValue(IBlockAccess iBlockAccess, int i, int i1, int i2) {
        return 15;
    }

    @Override
    public boolean preInit() {
        GameRegistry.registerBlock(this, ItemBlockBeacon.class, "Beacon");
        blockBeacon = new ItemStack(this, 1, 0);
        return true;
    }

    @Override
    public boolean initialize() {
        return true;
    }

    @Override
    public boolean postInit() {
        return true;
    }

    @Override
    public TileEntity createNewTileEntity(World world, int metadata) {
        return new TileEntityBeacon();
    }

    @Override
    public boolean isOpaqueCube() {
        return false;
    }

    @Override
    public boolean renderAsNormalBlock() {
        return false;
    }

    @Override
    public int getRenderType() {
        return -1;
    }

    @Override
    public void getBlockInfo(IBlockAccess world, int x, int y, int z, ForgeDirection dir, EntityPlayer player, List<IChatComponent> list, boolean b) {
        TileEntity tile = world.getTileEntity(x, y, z);
        if(tile instanceof ITileInfo) {
            ((ITileInfo)tile).getTileInfo(list, dir, player, b);
        }
    }
}

File: src/main/kotlin/info/devels/jtte/blocks/BlockBeacon.kt
package info.devels.jtte.blocks

import cofh.core.block.BlockCoFHBase
import cpw.mods.fml.common.registry.GameRegistry
import info.devels.jtte.JTTE
import info.devels.jtte.entities.TileEntityBeacon
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.IBlockAccess
import net.minecraft.world.World
import java.util.*


class BlockBeacon : BlockCoFHBase(Material.iron) {
    init {
        setHardness(5.0f)
        setResistance(100f)
        setStepSound(Block.soundTypeStone)
        setCreativeTab(JTTE.tab)
        setBlockTextureName("obsidian")
        setBlockName("jtte.beacon")

        setBlockBounds(0f, 0f, 0f, 1f, 1.8f, 1f)
    }

    override fun getLightValue(iBlockAccess: IBlockAccess, i: Int, i1: Int, i2: Int) = 15

    override fun preInit(): Boolean {
        GameRegistry.registerBlock(this, ItemBlockBeacon::class.java, "Beacon")
        blockBeacon = ItemStack(this, 1, 0)
        return true
    }

    override fun initialize(): Boolean {
        return true
    }

    override fun postInit(): Boolean {
        return true
    }

    override fun createNewTileEntity(world: World, metadata: Int): TileEntity {
        return TileEntityBeacon()
    }

    override fun isOpaqueCube(): Boolean {
        return false
    }

    override fun renderAsNormalBlock(): Boolean {
        return false
    }

    override fun getRenderType(): Int {
        return -1
    }

    override fun dismantleBlock(p0: EntityPlayer?, p1: NBTTagCompound?, p2: World?, p3: Int, p4: Int, p5: Int, p6: Boolean, p7: Boolean): ArrayList<ItemStack> {
        return arrayListOf(blockBeacon);
    }

    override fun getItemDropped(metadata: Int, seed: Random, asd: Int): Item? {
        return blockBeacon.item;
    }

    companion object {
        lateinit var blockBeacon: ItemStack
    }
}
